sensorium/training/trainers.py

```diff
@@ -274,22 +296,29 @@ def full_objective(model, dataloader, data_key, *args, **kwargs):
                     features = readout.features.cpu().detach().squeeze().T.numpy()
```

Comment:
@Nisone2000 why do you do detach here? It would stop gradients through the features for clustering loss. Is it only for the initialization purposes?

Reply:
Yes that's just for the initialization. I also have with torch.no_grad() around this so it's not necessary anymore. I can remove it.